provider/cloudflare/cloudflare.go

```diff
@@ -707,7 +765,19 @@ func groupByNameAndTypeWithCustomHostnames(records []cloudflare.DNSRecord, chs [
 		}
 		targets := make([]string, len(records))
 		for i, record := range records {
-			targets[i] = record.Content
+			if record.Type == "SRV" {
+				if dataMap, ok := record.Data.(map[string]interface{}); ok {
```

Comment:
Not covered with tests
![Screenshot 2025-02-27 at 07 25 23](https://github.com/user-attachments/assets/9bf86161-5cb6-4a04-a947-e8ad080acc03)